python/whylogs/core/validators/condition_validator.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+from itertools import chain
+from typing import Any, Callable, Dict, List
+
+from whylogs.core.metrics.condition_count_metric import Condition
+from whylogs.core.preprocessing import PreprocessedColumn
+from whylogs.core.validators.validator import Validator
+
+
+@dataclass
+class ConditionValidator(Validator):
+    def __init__(
+        self, name: str, conditions: Dict[str, Condition], actions=List[Callable[[str, str, Any], None]]
+    ) -> None:
+        self.conditions = conditions
+        self.name = name
+        self.actions = actions
+        self.total: int = 0
+        self.failures: Dict[str, int] = dict()
+
+        if "total" in self.conditions.keys():
+            raise ValueError("Condition cannot be named 'total'")
+
+        if "total_failures" in self.conditions.keys():
+            raise ValueError("Condition cannot be named 'total_failures'")
+
+        for cond_name in self.conditions.keys():
+            if cond_name not in self.failures:
+                self.failures[cond_name] = 0
+
+    def columnar_validate(self, data: PreprocessedColumn) -> None:
+        count = 0
+        count_failures = 0
+        for x in list(chain.from_iterable(data.raw_iterator())):
+            count += 1
+            for cond_name, condition in self.conditions.items():
+                try:
+                    if not condition.relation(x):
+                        self.failures[cond_name] += 1
+                        count_failures += 1
+                        for action in self.actions:
+                            action(self.name, cond_name, x)
+                except:  # noqa
+                    pass
```

Comment:
it is surprising to have an exception be "passing" the condition.

What if we remove the try/except block?

or if we keep it, make the except case count as a failure by default.

Reply:
I agree, removed it in the latest commit